src/components/DatePicker/DatePickerPopover.jsx

```diff
@@ -42,8 +42,6 @@ export const DatePickerPopover = ({
     };
 
     const handleClickOutside = () => {
-        // Revert back to the original value because the user didn't apply the changes
-        onChange(originalValue);
```

Comment:
Removed onChange handler from click outside as it casuses side-effects on the CreatePurchase flow and no longer needed as theres no Apply Button